src/client/app/strings/index.ts

```diff
@@ -50,7 +50,17 @@ export function Strings(): ClientStrings {
 }
 
 export function getDisplayLanguage() {
-    return (getRawDisplayLanguage() || 'en').toLowerCase();
+    let rawDisplayLanguage = getRawDisplayLanguage();
+    if (rawDisplayLanguage) {
+        rawDisplayLanguage = rawDisplayLanguage.toLowerCase();
+        for (let language of availableLanguages) {
+            if (rawDisplayLanguage === language.value) {
+                return rawDisplayLanguage;
+            }
+        }
+    }
+
+    return (rawDisplayLanguage || 'en').substr(0, 2);
```

Comment:
It is not a concern because we first check if the full language/region is translated, if so we choose that language because it exists in the list. If the region is not translated, we look up by primary language.